pgjdbc/src/main/java/org/postgresql/core/VisibleBufferedInputStream.java

```diff
@@ -350,4 +350,18 @@ public void setTimeoutRequested(boolean timeoutRequested) {
   public InputStream getWrapped() {
     return wrapped;
   }
+
+  public int receiveInteger2() throws IOException {
+    if (ensureBytes(2)) {
+      return (buffer[index++] & 0xFF) << 8 | buffer[index++] & 0xFF;
+    }
+    throw new EOFException("End of stream reached while trying to read integer2");
+  }
+
+  public int receiveInteger4() throws IOException {
+    if (ensureBytes(4)) {
+      return (buffer[index++] & 0xFF) << 24 | (buffer[index++] & 0xFF) << 16 | (buffer[index++] & 0xFF) << 8 | buffer[index++] & 0xFF;
+    }
+    throw new EOFException("End of stream reached while trying to read integer4");
+  }
```

Comment:
Would you please check out https://google.github.io/styleguide/javaguide.html#s3.4.2-ordering-class-contents ?

> new methods are not just habitually added to the end of the class

Have you considered adding the methods closer to the existing `read()` ones?

Then the naming could be like `readInt2()`, `readInt4()` (see `PgBufferedOutputStream#writeInt4`)